src/scene/assetmanager.cpp

```diff
@@ -24,106 +24,751 @@
+    }
+    _onDeinitializeFunctionRefs[asset].clear();
+
+    //
+    // Tear down asset lua table
+    const int top = lua_gettop(*_luaState);
+    // Push the global table of AssetInfos to the lua stack.
+    lua_rawgeti(*_luaState, LUA_REGISTRYINDEX, _assetsTableRef);
+    const int globalTableIndex = lua_gettop(*_luaState);
+
+    ghoul::lua::push(*_luaState, ghoul::lua::nil_t());
+
+    // Clear entry from global asset table (pushed to the Lua stack earlier)
+    std::string path = asset->path().string();
+    lua_setfield(*_luaState, globalTableIndex, path.c_str());
+    lua_settop(*_luaState, top);
+
+    
+    const auto it = std::find_if(
+        _assets.begin(),
+        _assets.end(),
+        [&asset](const std::unique_ptr<Asset>& a) { return a.get() == asset; }
+    );
+    if (it != _assets.end()) {
+        // Instead of deleting the asset directly, we are moving it into a to-delete queue
+        // as this unloadAsset function is called from the asset that we are manipulating
+        // right now. And deleting the asset while we are in a function of the same asset
+        // might be painful
+        _toBeDeleted.push_back(std::move(*it));
+        _assets.erase(it);
+    }
+}
+
+void AssetManager::setUpAssetLuaTable(Asset* asset) {
+    // Set up Lua table:
+    // AssetInfo
+    // |- Exports (table<name, exported data>)
+    // |- Asset
+    // |  |- localResource
+    // |  |- syncedResource
+    // |  |- require
+    // |  |- exists
+    // |  |- export
+    // |  |- onInitialize
+    // |  |- onDeinitialize
+    // |  |- directory
+    // |- Dependants (table<dependant, Dependency dep>)
+    //
+    // where Dependency is a table:
+    // Dependency
+    // |- onInitialize
+    // |- onDeinitialize
+
+    const int top = lua_gettop(*_luaState);
+
+    // Push the global table of AssetInfos
+    lua_rawgeti(*_luaState, LUA_REGISTRYINDEX, _assetsTableRef);
+    const int globalTableIndex = lua_gettop(*_luaState);
+
+    // Create a AssetInfo table for the current asset
+    lua_newtable(*_luaState);
+    const int assetInfoTableIndex = lua_gettop(*_luaState);
+
+    // Register empty Exports table for the current asset
+    // (string => exported object)
+    lua_newtable(*_luaState);
+    lua_setfield(*_luaState, assetInfoTableIndex, ExportsTableName);
+
+    // Create Asset table
+    // (string => Lua functions)
+    lua_newtable(*_luaState);
+    const int assetTableIndex = lua_gettop(*_luaState);
+
+    // Register local resource function
+    // string localResource(string path)
+    ghoul::lua::push(*_luaState, asset);
+    lua_pushcclosure(
+        *_luaState,
+        [](lua_State* L) {
+            ZoneScoped
+            
+            Asset* asset = ghoul::lua::userData<Asset>(L, 1);
+            ghoul::lua::checkArgumentsAndThrow(L, 1, "lua::localResourceLua");
+
+            std::string name = ghoul::lua::value<std::string>(L);
+            std::string path = fmt::format(
```

Comment:
Can this be another good place to use the `operator/` of the `std::filesystem::path`? :) E.g: 
```
std::filesystem::path path = asset->path().parent_path() / name;
ghoul::lua::push(L, path.string());
```

Reply:
I really don't like the `/` but I guess I lost that battle 😅